src/containers/barray.mli

```diff
@@ -0,0 +1,213 @@
+
+(* This file is free software, part of containers. See file "license" for more details. *)
+
+type 'a t = 'a array
+
+val empty : 'a t
+
+val isEmpty: 'a t -> bool
+
+external make : int -> 'a -> 'a t = "caml_make_vect"
+(** [Barray.make n x] returns a fresh t of length [n],
+    initialized with [x].
+    All the elements of this new t are initially
+    physically equal to [x] (in the sense of the [==] predicate).
+    Consequently, if [x] is mutable, it is shared among all elements
+    of the t, and modifying [x] through one of the t entries
+    will modify all other entries at the same time.
+    Raise [Invalid_argument] if [n < 0] or [n > Sys.max_array_length].
+    If the value of [x] is a floating-point number, then the maximum
+    size is only [Sys.max_array_length / 2].*)
+
+val init : int -> (int -> 'a) -> 'a t
```

Comment:
I stuck with 'init' in imm.re. I don't think it matters much though either way in this case.